inotify.go

```diff
@@ -175,6 +167,20 @@ func (w *Watcher) readEvents() {
 			return
 		}
 
+		ok, errno = w.poller.wait()
+		if errno != nil {
+			select {
```

Comment:
I'm starting to wonder if we should have a helper function to send an `error` down the channel or err/abort if done.